diff --git a/node_modules/@httptoolkit/httpolyglot/dist/index.js b/node_modules/@httptoolkit/httpolyglot/dist/index.js
index 935ac1b..ca44fbf 100644
--- a/node_modules/@httptoolkit/httpolyglot/dist/index.js
+++ b/node_modules/@httptoolkit/httpolyglot/dist/index.js
@@ -31,12 +31,9 @@ class Server extends net.Server {
             tlsConfig = configOrServerOrListener;
             requestListener = listener;
         }
-        // We bind the request listener, so 'this' always refers to us, not each subserver.
-        // This means 'this' is consistent (and this.close() works).
-        const boundListener = requestListener.bind(this);
         // Create subservers for each supported protocol:
-        this._httpServer = new http.Server(boundListener);
-        this._http2Server = http2.createServer({}, boundListener);
+        this._httpServer = new http.Server(requestListener);
+        this._http2Server = http2.createServer({}, requestListener);
         if (tlsServer) {
             // If we've been given a preconfigured TLS server, we use that directly, and
             // subscribe to connections there
